### ISSUE TITLE
NRE in One-To-Many result mapping

### ISSUE THREAD

**Issue by jeroenbu** [jonwagner/Insight.Database#219]
Hi,

I've defined a method in my interface IBeer as follows:

```
       [Recordset(1, typeof(Glass), IsChild = true, GroupBy = "FKBeer", Id = "ID")]
        Beer SelectBeer(int beerId);
```

The stored procedure SelectBeer produces two resultsets, namely one Beer, and any number of Glass (0, 1, or more). IEnumerable<Glass> Glasses is a property of Beer.

This goes well, as long as there's a Beer returned. If the result of the sp has no beer (and in this case also, no Glasses), a NullReferenceException occurs (I suspect because Insight tries to map the empty resultset of IEnumerable<Glass> to the property Glasses of the non-existent Beer.

Am I doing something wrong in the setup of the interface? 

Thx,

Jeroen

**Comment by jonwagner** [comment]
That's probably a bug.

**Comment by jonwagner** [comment]
Let's see if we can get a branch "issue-291" with a test case for this in MappingTests.cs. Once we have a stack trace, it should be an easy fix in ChildRecordReader or ChildMapper.

**Comment by Jaxelr** [comment]
Addressed on PR #282

**Comment by jonwagner** [comment]
Fixed in 5.2.9
